src/main/java/nextstep/optional/Users.java

```diff
@@ -12,12 +14,29 @@ public class Users {
             new User("jk", 40),
             new User("honux", 45));
 
+    /**
+     nextstep.optional.Users의 getUser() 메소드를 자바 8의 stream과 Optional을 활용해 구현한다.
+     자바 8의 stream과 Optional을 사용하도록 리팩토링한 후 UsersTest의 단위 테스트가 통과해야 한다.
+
+     힌트
+     Guide To Java 8 Optional 문서를 참고해 Optional 사용 방법을 익힌다.
+     Optional의 orElse() 메소드 활용해 구현한다.
+     */
     User getUser(String name) {
-        for (User user : users) {
-            if (user.matchName(name)) {
-                return user;
-            }
+        Optional<User> optionalUser = users.stream()
+                .filter(user -> user.matchName(name))
+                .findFirst();
+
+        if(optionalUser.isPresent()){
+            return optionalUser.get();
         }
-        return DEFAULT_USER;
+        return optionalUser.orElse(DEFAULT_USER);
```

Comment:
로직을 분리하지 않고 이어서 작성해볼 수 있지 않을까 생각합니다.
```suggestion
        return users.stream()
                .filter(user -> user.matchName(name))
                .findFirst()
                .orElse(DEFAULT_USER);
```

Reply:
```    User getUser(String name) {
        Optional<User> optionalUser = Optional.of(users.stream()
                .filter(user -> user.matchName(name))
                .findFirst()
                .orElse(DEFAULT_USER));
        return optionalUser.get();
    }

적용하여 처리하였습니다.
더 나은 방법 있으면 가이드 부탁드려요~

Reply:
제시해드린 코드를 `Optional.of()`로 감싸서 처리하신 이유가 있으실까요?